CMakeLists.txt

```diff
@@ -75,6 +75,8 @@ endif()
 
 if( CXX_VERSION_STRING MATCHES "clang" )
   message( STATUS "Use hip-clang to build for amdgpu backend" )
+# set ( CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Xclang -fallow-half-arguments-and-returns" )
+  set ( CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -D__HIP_HCC_COMPAT_MODE__=1" )
```

Comment:
These flags should be added to the `hip::host` and `hip::device` targets in hip.

Reply:
this option is a workaround for rocblas. it allows rocblas to use hcc workitem functions instead of HIP builtin variables. other libraries do not need this option.